uqcsbot/channels.py

```diff
@@ -109,36 +126,67 @@ async def listchannels(self, ctx: commands.Context):
 
     @commands.command()
     @commands.has_permissions(manage_channels=True)
-    async def addjoinchannel(self, ctx: commands.Context, channel: discord.TextChannel):
+    async def addjoinchannel(self, ctx: commands.Context, channel: discord.TextChannel, emoji):
         """ Sets a channel to be joinable with UQCSbot. """
+        if not self._valid_emoji(emoji):
+            await ctx.send("Please select an emoji found within this server.")
+            return
+
         db_session = self.bot.create_db_session()
+        taken = db_session.query(Channel).filter(Channel.id != channel.id,
+                                                 Channel.emoji == emoji).first()  
+        if taken:
+            db_session.close()
+            await ctx.send("Please select an emoji that isn't already in use.") 
+            return
 
         existing = db_session.query(Channel).filter(Channel.id == channel.id).one_or_none()
         if existing:
             existing.joinable = True
+            existing.emoji = emoji
         else:
-            db_session.add(Channel(id=channel.id, name=channel.name, joinable=True))
-
+            db_session.add(Channel(id=channel.id, name=channel.name, joinable=True, emoji=emoji))
         db_session.commit()
         db_session.close()
+
         await ctx.send(f"{channel.mention} was added as a joinable channel.")
+        react_message = await ctx.fetch_message(self.message_id)
+        await self.update_message(react_message)
 
     @commands.command()
     @commands.has_permissions(manage_channels=True)
     async def removejoinchannel(self, ctx: commands.Context, channel: discord.TextChannel):
         """ Sets a channel to be not joinable with UQCSbot. """
         db_session = self.bot.create_db_session()
-
         try:
             existing = db_session.query(Channel).filter(Channel.id == channel.id).one()
             existing.joinable = False
         except NoResultFound:
             await ctx.send(f"There was no record for {channel.mention}. The channel is not currently joinable.")
             return
-
         db_session.commit()
         db_session.close()
+
         await ctx.send(f"{channel.mention} was removed as a joinable channel.")
+        react_message = await ctx.fetch_message(self.message_id)
+        await self.update_message(react_message)
+
+    @commands.command(hidden=True)
+    @commands.has_permissions(manage_channels=True)
```

Comment:
Making this hidden hides it for all users, including committee. I'd leave it not hidden as it will automatically be hidden from help for anyone who can't manage channels.

Reply:
Oh true, I hadn't considered it would be invisible for committee as well. Thanks.